#Parivastan (Hindi: परिवर्तन)is a SNP calling pipeline -- Under development
#Jitendra / jnarayan81@gmail.com


#Index the fasta
bwa index my.fasta

#Prepare for the alignments
bwa aln [opts] my.fasta my.fastq > my.sai

#Store in SAM format
bwa samse my.fasta my.sai my.fastq > my.sams

#SAM to BAM 
samtools view -S -b my.sam > my.bam

#Sort BAM for SNP calling
samtools sort my.bam my-sorted

#Index the genome assembly
samtools faidx my.fasta

#Run 'mpileup' to generate VCF format
samtools mpileup -g -f my.fasta my-sorted-1.bam my-sorted-2.bam my-sorted-n.bam > my- raw.bcf

#Call SNPs...
bcftools view -bvcg my-raw.bcf > my-var.bcf

#Filter SNPs
bcftools view my.var.bcf | vcfutils.pl varFilter - > my.var-final.vcf

*/
Usage:   vcfutils.pl varFilter [options] <in.vcf>
 Options: -Q INT
         -d INT
         -D INT
         -a INT
         -w INT
         -W INT
minimum RMS mapping quality for SNPs [10]
minimum read depth [2]
maximum read depth [10000000]
minimum number of alternate bases [2]
SNP within INT bp around a gap to be filtered [3]
window size for filtering adjacent gaps [10]
         -1 FLOAT  min P-value for strand bias (given PV4) [0.0001]
         -2 FLOAT  min P-value for baseQ bias [1e-100]
         -3 FLOAT  min P-value for mapQ bias [0]
         -4 FLOAT  min P-value for end distance bias [0.0001]
         -e FLOAT  min P-value for HWE (plus F<0) [0.0001]
         -p        print filtered variants
Note: Some of the filters rely on annotations generated by SAMtools/BCFtools.
*/`
